# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/CacheIRCompiler.cpp
# Commit: bcd63afc2d34
# Full Hash: bcd63afc2d3449830cfe6f51c03373f890ef6435
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2023-07-06 04:06:52
# Regressor Bug: 1837620
# File Overlap Count: 1
# Description:
#   Bug 1837620 - Part 6: Make edges for multiple shape guard weak too r=sfink
#   
#   To make the edges for the multiple shape guard weak we need a way to find and
#   update the objects used to hold these edges. The simplest way to do this is to
#   add a vector in the zone.  I don't believe that there are not huge numbers of
# ==============================================================================

diff -r 24e0b10be67a -r bcd63afc2d34 js/src/jit/CacheIRCompiler.cpp
--- a/js/src/jit/CacheIRCompiler.cpp	Wed Jul 05 16:58:38 2023 +0000
+++ b/js/src/jit/CacheIRCompiler.cpp	Wed Jul 05 16:58:38 2023 +0000
@@ -1229,10 +1229,11 @@
         TraceEdge(trc, &stubInfo->getStubField<T, GetterSetter*>(stub, offset),
                   "cacheir-getter-setter");
         break;
-      case StubField::Type::JSObject:
+      case StubField::Type::JSObject: {
         TraceEdge(trc, &stubInfo->getStubField<T, JSObject*>(stub, offset),
                   "cacheir-object");
         break;
+      }
       case StubField::Type::WeakObject:
         if (ShouldTraceWeakEdgeInStub<T>(trc)) {
           TraceEdge(trc, &stubInfo->getStubField<T, JSObject*>(stub, offset),